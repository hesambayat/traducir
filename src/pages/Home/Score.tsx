import { useModal } from '../../context/modal'

interface ScoreProps {
  total: number
  remaining: number
}

const Score = ({ total, remaining }: ScoreProps) => {
  const { setModal } = useModal()

  return (
    <header>
    <button className="home__score" onClick={setModal}>
      <span>?</span> You got {total - remaining}/{total}
    </button>
  </header>
  )
}

export default Score
