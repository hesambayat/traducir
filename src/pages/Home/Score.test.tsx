import { render } from '@testing-library/react'
import Score from './Score'

describe('renders home page feature / score', () => {
  const { asFragment } = render(<Score total={9} remaining={0} />)
  const dom = asFragment()
  test('should match the snapshot', () => {
    expect(dom).toMatchSnapshot()
  })
})
