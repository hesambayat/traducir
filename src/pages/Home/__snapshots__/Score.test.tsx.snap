// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page feature / score should match the snapshot 1`] = `
<DocumentFragment>
  <header>
    <button
      class="home__score"
    >
      <span>
        ?
      </span>
       You got 9/9
    </button>
  </header>
</DocumentFragment>
`;
